src/main/java/io/vertx/core/net/impl/SSLHelper.java

```diff
@@ -410,9 +407,9 @@ private SslHandler createHandler(SSLEngine engine, boolean client) {
           break;
         }
       }
-    } else if (verifyHost) {
+    } else if (!endpointIdentificationAlgorithm.isEmpty()) {
```

Comment:
Added Objects.requireNonNull for the setter, I think this is preferred over null checking everywhere the value might be used.